Doc/c-api/type.rst

```diff
@@ -112,4 +103,125 @@ Type Objects
    Callers will typically cast the result pointer into the appropriate
    function type.
 
+   See :c:member:`PyType_Slot.slot` for possible values of the *slot* argument.
+
    .. versionadded:: 3.4
+
+
+Creating Heap-Allocated Types
+-----------------------------
+
+Traditionally, types defined in C code are *static*, that is,
+a ``static PyTypeObject`` structure is defined directly in code
+and initialized using :py:func:`PyType_Ready`.
+
+This results in types that are limited relative to types defined in Python:
+
+* Static types are limited to one base, i.e. they cannot use multiple
+  inheritance.
+* Static type objects (but not necessarily their instances) are immutable.
+  It is not possible to add or modify the type object's attributes from Python.
+* Static type objects are shared across
+  :ref:`sub-interpreters <sub-interpreter-support>`, so they should not
+  include any subinterpreter-specific state.
+
+Also, since *PyTypeObject* is not part of the stable ABI, any extension modules
```

Comment:
Actually, extension modules using static types should be compatible with newer (and may be even older) Python versions. New fields are added to the end of `PyTypeObject` and structures like `PyNumberMethods` and should be used only when corresponded type flag is set (see `PyType_HasFeature()`).

This is in theory.